src/main/java/net/starlark/java/syntax/Expression.java

```diff
@@ -46,15 +46,20 @@ public enum Kind {
     UNARY_OPERATOR,
   }
 
-  Expression(FileLocations locs) {
+  private final Kind kind;
```

Comment:
This needs a comment recording that the materialized field is a CPU optimization, not a missed opportunity to save space.

Reply:
Resolved in the internal merge.